### ISSUE TITLE
Duplicate Token `<s>` in Tokenizer Encoded Token ids

### ISSUE THREAD

**Issue by zxybazh** [vllm-project/vllm#2899]
When working on tokenizer result for `llama-2-7b-chat-hf` model, I noticed that the `prompt_token_ids` generated in [this place](https://github.com/vllm-project/vllm/blob/5f08050d8d0bfcdaced0fe706cdfc9e311e0f263/vllm/engine/llm_engine.py#L385C13-L385C29) would generate an extra token `<s>` in the beginning of the sentence.

For example for the follow prompt `<s>[INST] what is the color of the snow? [/INST]` , hf tokenizer can directly tokenize it to
```
['<s>', '▁[', 'INST', ']', '▁what', '▁is', '▁the', '▁color', '▁of', '▁the', '▁snow', '?', '▁[', '/', 'INST', ']']
[1, 518, 25580, 29962, 825, 338, 278, 2927, 310, 278, 15007, 29973, 518, 29914, 25580, 29962]
```
but for the very same prompt vllm would generate tokenized prompt ids as follows
```
[1, 1, 518, 25580, 29962, 825, 338, 278, 2927, 310, 278, 15007, 29973, 518, 29914, 25580, 29962]
```
which has an extra token `1`, aka `<s>` in the beginning.

Looking forward to have someone help me confirm if this is designated behaviour or caused by some of the model options.

**Comment by CatherineSue** [comment]
I think this is because in llama2-7b-chat-hf's[ tokenizer_config.json](https://huggingface.co/meta-llama/Llama-2-7b-chat-hf/blob/main/tokenizer_config.json#L2), it sets `add_bos_token` to true.

**Comment by zxybazh** [comment]
Thanks for the clarification, in that case it should be designated behaviour.

**Comment by CatherineSue** [comment]
also a fix is out #4688
